x/lockup/keeper/lock.go

```diff
@@ -281,6 +286,43 @@ func (k Keeper) UnlockPeriodLockByID(ctx sdk.Context, LockID uint64) (*types.Per
 	return lock, err
 }
 
+// AddTokensToLock locks more tokens into a lockup
+func (k Keeper) AddTokensToLock(ctx sdk.Context, owner sdk.AccAddress, lockID uint64, coins sdk.Coins) (*types.PeriodLock, error) {
+	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return nil, err
+	}
+	if lock.Owner != owner.String() {
+		return nil, types.ErrNotLockOwner
+	}
+	if err := k.bk.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, coins); err != nil {
+		return nil, err
+	}
+	lock.Coins = lock.Coins.Add(coins...)
+
+	// store lock object into the store
+	store := ctx.KVStore(k.storeKey)
+	bz, err := proto.Marshal(lock)
+	if err != nil {
+		return nil, err
+	}
+	store.Set(lockStoreKey(lock.ID), bz)
```

Comment:
Shouldn't there be an existing method for saving the lock to the store?

Reply:
If theres no existing method, there should at least be a way to reduce the code duplication here

Reply:
Probably, will reduce duplications

Reply:
Reduced duplication on this commit https://github.com/osmosis-labs/osmosis/pull/381/commits/c39118acb6688bae2e99189c3b988d2aa6f86b28